pkg/synchronization/controller.go

```diff
@@ -456,7 +461,25 @@ func (c *Controller) createOrUpdateServiceMetadata(logger *zap.Logger, ns *core_
 	return false, nil
 }
 
-func (c *Controller) getServiceData(user auth.OptionalUser, name voyager.ServiceName) (*creator_v1.Service, error) {
+func (c *Controller) getServiceData(user auth.OptionalUser, name voyager.ServiceName, allowCached bool) (*creator_v1.Service, error) {
```

Comment:
TODO: introduce a mutex to make sure there are no concurrent requests between a controller from the queue and a polling goroutine

Reply:
Actually I think we can only fetch data from the cache for namespace sync (the item is guaranteed to be present in cache, otherwise SD cannot be created without RBAC being setup by a polling loop) and use `sync.RWLock`.
Will rewrite a bit.